test/spinxsk/spinxsk.c

```diff
@@ -2298,7 +2422,21 @@ QueueWorkerFn(
             // Let datapath thread/s pump datapath for set duration.
             //
             TraceVerbose("q[%u]: letting datapath pump", queue->queueId);
-            WaitForSingleObject(stopEvent, 500);
+
+            if (queue->fnmp != NULL) {
+                //
+                // Inject packets from FNMP if it is available
+                //
+                for (int i = 0; i < 10; ++i) {
+                    UINT8 numFrameInjected = InjectFnmpRxPacket(queue->fnmp);
+                    TraceVerbose("q[%u]: %d frames injected through FNMP", queue->queueId, numFrameInjected);
+                    if (WAIT_OBJECT_0 == WaitForSingleObject(stopEvent, 50)) {
```

Comment:
i wouldn't bother checking that here

Reply:
oh, yeah, this is the wrong place to do this, i think

Reply:
I considered moving this in another thread, but we need only a single worker doing this and this one is idle waiting for the datapath threads, so it can do it concurrently with the rest.
Waiting on the event is more about the 50ms delay than stopEvent, it was any easy way to spread frames until the proper generation is done.

Reply:
my concern is that more generally, we do not want to introduce artificial sequencing of independent fuzzing actions. for example, fuzzing the FNMP packet data _during_ socket setup is interesting, but would be missed if fuzzing only runs on this thread.

it is also inadequate because we want to test XDP with multiple concurrent receive indications per queue, which won't work if the datapath fuzzer is running per-queue.